halo2-base/src/gates/range.rs

```diff
@@ -472,6 +494,33 @@ impl<F: ScalarField> RangeChip<F> {
     pub fn default(lookup_bits: usize) -> Self {
         Self::new(RangeStrategy::Vertical, lookup_bits)
     }
+
+    /// Bitwise right rotate a by <bit> bits. This function should never be called directly
+    /// because const bitwise rotation must be determined at compile time.
+    ///
+    /// Assumes 'a' is a <num_bits> bit integer and 0 < <num_bits> <= 128.
```

Comment:
the 128 bit limit could be removed and replaced by `F::CAPACITY`, if desired, by using `fe_to_biguint` and `biguint_to_fe`. 

Optional suggestion